### PR TITLE
Use immutable data structures for applab screen reducer

### PR DESCRIPTION
**Pull request by pcardune**
cc @Bjvanminnen and @islemaster

### CODE REVIEW

File: apps/src/applab/designMode.js [hunk 1]
```diff
@@ -30,18 +30,18 @@ var ApplabInterfaceMode = applabConstants.ApplabInterfaceMode;
...
-    screens: {}
-  };
+  var state;
   store.subscribe(function () {
     var lastState = state;
     state = store.getState();
 
-    if (state.screens.currentScreenId !== lastState.screens.currentScreenId) {
-      renderScreens(state.screens.currentScreenId);
+    if (!lastState ||
```
Comment (bcjordan) [inline apps/src/applab/designMode.js:40]: One option that might help a tiny bit with scan-ability of these lines:

```
const screenChanged = [boolean logic];
if (screenChanged) {
```

(no strong feeling on this, though)

**Comment by bcjordan** [comment]
LGTM!

**Comment by Bjvanminnen** [comment]
Generally lgtm. Any thoughts on whether we could/should have a way of clearly indicating which reducers expose Immutable objects vs. JS objects?

**Comment by islemaster** [comment]
image

**Comment by pcardune** [comment]
@Bjvanminnen perhaps a naming convention? I would say prefix everything with `immutable` but that is too long. Perhaps postfix with the specific immutable type? So this would become `state.screensMap.get(...)`

**Comment by Bjvanminnen** [comment]
Another option (that I don't think I like) would be to put immutable reducers in their own subtree, i.e.

```
state = {
  instructions: { ... },
  immutable: {
    screens: Immutable.Map({...}),
    somethingElse: Immutable.Map({...})
 }
};
```

It's also possible that we don't need to do anything here, and it will be reasonably clear just by looking at the reducer or other consumers of the reducer whether or not it's using Immutable.

**Comment by pcardune** [comment]
going to just leave this in my applab import branch

**Pull request closed without merging**